src/huggingface_hub/utils/_hf_folder.py

```diff
@@ -0,0 +1,64 @@
+# coding=utf-8
+# Copyright 2022-present, the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Contain helper class to retrieve/store token from/to local cache."""
+import os
+from pathlib import Path
+from typing import Optional
+
+
+class HfFolder:
```

Comment:
Are we ok with moving the location of `HfFolder`? Any backward compatibility issues we need to keep an eye at?

Reply:
We could reimport it in the `hf_api.py` file just to make sure BC is respected. I agree it's important given that in this libary we have private modules (`_deprecation.py`, `_headers.py`, ...) which sends the signal that public modules should have stable import APIs

Reply:
Nice catch @osanseviero , thanks for reporting.

Actually it would have break [here](https://github.com/huggingface/transformers/blob/c7fd28999fb8a03f4839ab5402a7c32997ded0fa/src/transformers/commands/user.py#L19) in `transformers`: . I made the re-import in `hf_api.py` as suggested by @LysandreJik.

For the record, I needed to remove `HfFolder` from `hf_api.py` to avoid circular imports between `utils._headers.py` and `hf_api.py`.